refactor: TestWallet API cleanup pt. 1

benesjan opened the pull request:
Followup work of https://github.com/AztecProtocol/aztec-packages/pull/17228

In this PR I drop `getPXEInfo` and `getContracts` from `TestWallet` + add random docs improvements.

`getPXEInfo` was just a tech debt as it was only used to check if node has the same protocol contract addresses which is quite non-sensical because those addresses are no literally 1, 2, 3, 4, 5... I assume this check is there from time when they didn't have these hardcoded addresses.

`getContracts` was used just in one test and the check there was not really that important so I just dropped it.

benesjan commented:
* **#17380** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17380?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17325** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17325?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17295** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17295?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17303** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17303?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17283** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17283?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17283?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on docs/docs/developers/reference/environment_reference/cli_reference.md:
```diff
@@ -590,24 +590,6 @@ Options:
 - `--sponsored-fpc`: Populate genesis state with a testing sponsored FPC contract
 - `--accelerated-test-deployments`: Fire and forget deployment transactions, use in testing only (default: false)
 
-### deploy-l1-verifier
```
This command doesn't exist on the cli.

benesjan commented on docs/docs/developers/reference/environment_reference/cli_wallet_reference.md:
```diff
@@ -208,7 +208,7 @@ Generates a secret key and deploys an account contract. Uses a Schnorr single-ke
 - `--public-deploy`: Publishes the account contract instance (and the class, if needed). Needed if the contract contains public functions.
 - `-p, --public-key <string>`: Public key that identifies a private signing key stored outside of the wallet. Used for ECDSA SSH accounts over the secp256r1 curve.
-- `-n, --node-url <string>`: URL of the PXE (default: "http://host.docker.internal:8080")
+- `-n, --node-url <string>`: URL of Aztec Node (default: "http://host.docker.internal:8080")
```
Forgot to update in the last PR. These references seems like a pretty bad thing to have given that `aztec --help` is available and there it will not become stale.

benesjan commented on yarn-project/aztec.js/src/wallet/account_manager.ts:
```diff
@@ -142,7 +142,7 @@ export class AccountManager {
    */
   public async register(): Promise<AccountWithSecretKey> {
-    await this.pxe.registerContract({
+    await this.wallet.registerContract({
```
Random change when I was investigating what is remaining from allowing me to drop the ugly `getPxe` method on test wallet. That function can be dropped once we allow Wallet.registerContract to accept secretKey and partialAddress on the input.

benesjan commented on yarn-project/cli-wallet/src/cmds/check_tx.ts:
```diff
@@ -23,20 +23,14 @@ export async function checkTx(
     const receipt = await aztecNode.getTxReceipt(txHash);
     return receipt.status;
   } else {
-    await inspectTx(wallet, aztecNode, txHash, log, { includeBlockInfo: true });
```
Just did a minor cleanup of this func.

benesjan commented on yarn-project/test-wallet/src/wallet/test_wallet.ts:
```diff
@@ -50,12 +49,29 @@ export interface AccountData {
+   * When this flag is true, simulateTx constructs a request using a fake account (and accepts contract overrides
+   * on the input) and the PXE emulates kernel effects without generating kernel witnesses. When false, simulateTx
+   * defers to the standard simulation path.
+   */
```
Docs based on comment Grego sent on slack.

Grego pls check it's no ambiguous.

Thunkar replied:
The gas estimations not being accurate is not a given, I wouldn't say that in the comment. We aim for them to be accurate and usable, and for now they are

benesjan replied:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/17283/commits/d7dd3d0618662404533c9fd5b73569ab4d2b0f5f

benesjan commented on yarn-project/cli-wallet/src/cmds/index.ts:
```diff
@@ -444,14 +444,10 @@ export function injectCommands(
       const [secret, messageLeafIndex] = await bridgeL1FeeJuice(
         amount,
         recipient,
-        wallet,
```
Wallet no longer needs to be passed in as we can just obtain all the necessary info from node.

Thunkar commented on yarn-project/test-wallet/src/wallet/test_wallet.ts:
```diff
@@ -170,6 +186,8 @@ export abstract class BaseTestWallet extends BaseWallet {
     executionPayload: ExecutionPayload,
     opts: SimulateMethodOptions,
   ): Promise<TxSimulationResult> {
+    // In simulated-simulation mode we emulate kernel logic in TS. This is fast but can skew gas,
```
I just dropped this from my current PR, comment is unnecessary

benesjan replied:
Nuked it in https://github.com/AztecProtocol/aztec-packages/pull/17283/commits/3413e33250a3fab8d5a3d4a1496b0400d357e7bb

benesjan committed:
refactor: TestWallet API cleanup pt. 1

Followup work of https://github.com/AztecProtocol/aztec-packages/pull/17228

In this PR I drop `getPXEInfo` and `getContracts` from `TestWallet` + add random docs improvements.

`getPXEInfo` was just a tech debt as it was only used to check if node has the same protocol contract addresses which is quite non-sensical because those addresses are no literally 1, 2, 3, 4, 5... I assume this check is there from time when they didn't have these hardcoded addresses.

`getContracts` was used just in one test and the check there was not really that important so I just dropped it.
